### PR TITLE
Support `UIImage` in product image upload

### PR DESCRIPTION
**Pull request by jaclync**
Prerequisite for HACK Week pe5pgL-3Bl-p2#comment-2957

## Why

In order to implement image processing like background removal for product images, the app needs to support image type `UIImage` in addition to `PHAsset` from the camera or device photo library. Unlike `PHAsset` which contains camera metadata, the filename and alt text need to be provided manually for a `UIImage` after image processing. This requires some changes in how we export and upload media. The changes in this PR are not expected to affect the app behavior, and the `UIImage` use case will be in a separate PR.

## How

First, `ProductImageStatus.uploading` enum case's associated type has been changed from `PHAsset` to a new enum type `ProductImageAssetType` which now can be either `PHAsset` or `UIImage` (plus filename and alt text). This change results in changes everywhere `ProductImageStatus.uploading` is used to also accommodate `UIImage`.

Then, the core changes are in the Yosemite layer:

- `MediaUIImageExporter` was created for exporting a `UIImage` to `UploadableMedia`
- In `MediaExportService`, it uses the new `MediaUIImageExporter` when the asset is `UIImage`
- In `MediaImageExporter` where both `PHAsset` and `UIImage` exporters call to export from image `Data` to `UploadableMedia`, the alt text is DI'ed and set in the output `UploadableMedia`
- `UIImage` now conforms to `ExportableAsset`, and `MediaAction` now includes `filename` and `altText` for `UIImage` or if we want to support metadata editing for all image types

Networking layer:

In `MediaRemote.uploadMediaToWordPressSite`, the interface was updated to only take in one media item because the WP API doesn't support multiple items. The return type is also already returning one media item. In both `MediaRemote.uploadMedia` functions, the alt text is now included in the network request. The implementation differs when using the WPCOM and site API.

## Testing instructions

There's no use case for the `UIImage` asset type right now, the background removal feature is only available for iOS 17+ so we need to wait for Xcode 15 to be adopted in CI first. For confidence check to make sure the changes don't affect the current image upload:

- Log in to a store if needed
- Go to the Products tab
- Tap on an editable product
- Tap on the images header
- Tap `Add Photos`
- Select either `Choose from device` and pick some photos --> after confirming the selection, the images should be uploaded to the site's media library in core successfully and have a filename that's not UUID (unless that's the actual filename of the image)
- Tap `Save` --> after the images are uploaded, the product should include the uploaded images

---

- [x] @jaclync tests the camera flow
- [x] @jaclync tests the image upload flow in a JCP site 

---
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: Networking/Networking/Model/TaxRate.swift [hunk 1]
```diff
@@ -16,14 +16,14 @@ public struct TaxRate: Decodable, Equatable, GeneratedFakeable, GeneratedCopiabl
...
-    ///
-    public let state: String
-
     /// Tax rate country.
     ///
     public let country: String
 
+    /// Tax rate state.
+    ///
+    public let state: String
```
Comment (jaclync) [inline Networking/Networking/Model/TaxRate.swift:25]: Reason for this unrelated change: 

Previously, the property order doesn't match the initializer and thus some extra diffs from Copiable/Fakeable are required after running `rake generate` (this probably wasn't run when the changes were last made). With this change, the property order now matches the order in the initializer.

**Review by ThomazFB — APPROVED** [review]
Test scenario reproduction matches the PR description. Looks good!

Screenshot 2023-09-08 at 11 37 55
Screenshot 2023-09-08 at 11 43 55

File: Networking/Networking/Model/TaxRate.swift (hunk 1 above)
Comment (jaclync) [reply Networking/Networking/Model/TaxRate.swift:25]: cc @toupper about keeping the order of the properties and in the initializer consistent; @Ecarrion for the changes on `TaxRate`

Comment (toupper) [reply Networking/Networking/Model/TaxRate.swift:25]: Thanks for tagging and letting me know 👍
